ready to install ts2dart into the angular build

alexeagle opened the pull request.

PatrickJS committed:
refactor(forEach): change to for-of with iterable

rename: foreach -> for
rename: array -> iterable
update: DartParseTreeWriter
update: naive_infinite_scroll
update: todo
fix: tests in foreach_spec

Closes #919

Commit:
refactor(dart/transform): Remove `ngData`

Now that we have `Parser`, `ngData` is redundant & unnecessary.

Commit:
refactor(dart/transform): Minor logging changes

Enable easier testing by providing a null log implementation and a way
to use it.

Commit:
refactor(dart/transform): Test `directive_linker` as a unit

Formerly, it was tested only as a piece of the transformer pipeline. Add
its own directory and test the linker on its own.

alexeagle committed:
feature(build): Include ts2dart transpile step in the Angular build.

This only transpiles one package to start with: di/
It ensures that package transpiles without errors, so no one can
introduce non-TypeScript syntax.
Next step is to widen the task inputs to cover additional packages.

See design doc for the migration:
https://docs.google.com/document/d/14RJLhu6uuv7NchFkAb6PKzOOO0L7l3Z507eKWzkEUhQ/edit

A convenience task 'ts2dart' is added for developing ts2dart, and
it runs all of the angular code through the transpiler to collect errors.

ovr committed:
README - don't forget to build app before start

Don't forget to build app before start HTTP server

Closes #1014

divdavem committed:
fix(ViewContainer) removeChild called with null parent

In view_container.js, templateElement.parentNode can be null
when two template tags are nested in one another.
Accessing the parent node through view.nodes[0].parentNode fixes
the problem.

closes #997

Closes #999

vandycknick committed:
docs: annotations

vandycknick committed:
(docs) decorator events property

As from what i understand shouldn't the event property rather be events: https://github.com/angular/angular/blob/master/modules/angular2/src/core/annotations/annotations.js#L161

Closes #1018

marclaval committed:
feat(selector): support , for multiple targets

Fixes #867
Closes #1019

chalin committed:
chore(install+test): single cmd to full install/test & test JS w/o Dart

* `npm install` now does a full install; auxiliary installation steps
have been integrated into the `postinstall` script.
* Updated developer docs `DEVELOPER.md` accordingly; also added
instructions to dev docs for performing full tests (via `npm test`) --
same as those run on Travis.
* Reorg in tests so that JS tests can run without a Dart env.

Partly fixes #945 **under the assumption that when running JS tests
locally, `ChromeCanary` is the desired browser to use**. Note that CI
tests (Travis) still uses `DartiumWithWebPlatform` across the board
(Maybe because ChromeCanary isn't being installed?)

Fixes #1012.

Closes #1010

pkozlowski-opensource committed:
refactor(DirectiveParser): remove checks for missing directives

Based on the discussion in #776 we can't reliably check if a given
element has a particular property at the compilation time. As such
the existing algorithm detecting "missing" directives can't be used.

We need to see if there is a different / better algorithm or maybe
those checks need to be moved later in the process (runtime). Leaving
integration tests in place (disabled) so we can come back to the
topic after unblocking the situation.

This commit effectivelly reverts 94e203b9df6c4b79ba30f1f08fc54ff919f722e1

vsavkin committed:
refactor(formed): changed forms to use event and property setters instead of NgElement

vsavkin committed:
feat(forms): added value accessor for input=text

vsavkin committed:
refactor(forms): wrapped all validators into the Validator class

vicb committed:
feat(ShadowCss): Support the new deep combinator syntax >>>

fixes #990

ref http://dev.w3.org/csswg/css-scoping-1/#deep-combinator

Closes #1028

marclaval committed:
feat(compiler): support bindings for any attribute

Closes #1029

pkozlowski-opensource committed:
fix(ElementBinderBuilder): properly bind to web component properties

Fixes #776

Closes #1024

mfollett committed:
Fix which dependency is injected w/ current elem.

Docs for the "Injecting a directive from the current element" indicate that having a dependency of `dependency: Dependency` should cause the current element's dependency to be injected, but then uses the ID value from the parent element in the example.

Closes #1032

mhevery committed:
docs: annotations

Commit:
refactor(dart/transform): Use package:guinness in tests

`guinness` is a Dart port of Jasmine. Since the rest of Angular 2 uses
Jasmine, use it for the transformer too.

Closes #8

Closes #1037

Closes #1000

Commit:
style(dart/transform): Remove `src` from library directives

Conform to Angular 2 style by removing `src` from library directives.
Completed with:
```
find -name "*.dart" | xargs sed -i -e 's!library\(.*\)src\.\(.*\)!library \1\2!'
```

Closes #1005

Closes #1038

mprobst commented on gulpfile.js:
```diff
@@ -115,7 +116,10 @@ var CONFIG = {
+      dart: ['modules/**/*.js'],
+      // Migrating to TypeScript, one package at a time.
+      // See https://docs.google.com/document/d/14RJLhu6uuv7NchFkAb6PKzOOO0L7l3Z507eKWzkEUhQ/edit
+      ts2dart: ['modules/angular2/src/di/*.js', 'modules/angular2/test/di/*.js', 'modules/angular2/src/test_lib/*.js']
```
Are we going to include the output of this in the actual production build?

mprobst commented on gulpfile.js:
```diff
@@ -115,7 +116,10 @@ var CONFIG = {
-      dart: ['modules/**/*.js']
+      dart: ['modules/**/*.js'],
+      // Migrating to TypeScript, one package at a time.
+      // See https://docs.google.com/document/d/14RJLhu6uuv7NchFkAb6PKzOOO0L7l3Z507eKWzkEUhQ/edit
```
Maybe create a g.co or goo.gl link?

mprobst commented:
I think you'll have to rebase this, this has 9 commits going into it. And squash my commit, I presume that's the problem with the CLA bot.

vsavkin committed:
fix(view): fixed view instantiation to use the component template's change detector when creating BindingPropagationConfig

yjbanov committed:
chore(packaging): bump version to alpha.14

alexeagle commented:
No, not included in either dist/dart or dist/js, so I think we are throwing
the output away at present.
Squashed the commits and rebased, but there is a test failure so we need a
fix in the transpiler first...

On Mon, Mar 23, 2015 at 2:54 PM Martin Probst <notifications@github.com>
wrote:

> I think you'll have to rebase this, this has 9 commits going into it. And
> squash my commit, I presume that's the problem with the CLA bot.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/angular/pull/1061#issuecomment-85217844>.
>

vicb committed:
doc(test): add a comment on why tests are disabled

petebacondarwin committed:
chore(doc-gen): add specific template for displaying variable exports

Closes #1071

petebacondarwin committed:
chore(doc-gen): move `@publicModule` tag-def to base package

This prevents unwanted "unknown tag" warnings when generating the non-public docs.

petebacondarwin committed:
chore(doc-gen): ignore non-jsdoc style comments

Now the visitor will find the last jsdoc style comment (e.g. `/** jsdoc comment */`)
before the current code item, ignoring any inline style comments (e.g. `// inline comment`)
in between.

Closes #1072

vsavkin committed:
chore: bump up the version to 2.0.0-alpha.15

alexeagle commented:
This was merged in 43f4374944a2b7c7933968782ec0b362f93bd2fd
